### PR TITLE
Austenem/cat 765 select default template

### PR DESCRIPTION
**Pull request by austenem**
## Summary

This update selects the blank template by default in the "Create New Workspace" dialog.

## Design Documentation/Original Tickets

[CAT-765 JIRA ticket](https://hms-dbmi.atlassian.net/browse/CAT-765?atlOrigin=eyJpIjoiNGQ0MmMyMmFjNjNjNGNkZTllMzA2YTZlYTFjMDI3ZWIiLCJwIjoiaiJ9)

## Testing

The altered component appears in the "Create new workspace" dialog and the "Add templates" dialog (see documentation below).

Both dialogs were manually tested by ensuring that the blank template is only selected in the "Create new workspace" dialog and that the template is able to be deselected.

## Screenshots/Video
Screenshot 2024-07-30 at 5 10 44 PM
Screenshot 2024-07-30 at 5 11 22 PM

## Checklist

- [x] Code follows the project's coding standards
  - [x] Lint checks pass locally
  - [X] New `CHANGELOG-your-feature-name-here.md` is present in the root directory, describing the change(s) in full sentences.
- [ ] Unit tests covering the new feature have been added
- [X] All existing tests pass
- [X] Any relevant documentation in JIRA/Confluence has been updated to reflect the new feature
- [X] Any new functionalities have appropriate analytics functionalities added

## Additional Notes

There is an ongoing issue with a `useEffect` dependency array in the `SelectableTemplateGrid` component - I'll drop a comment about this below, but would appreciate some feedback about the best way to handle this.

### CODE REVIEW

File: context/app/static/js/components/workspaces/SelectableTemplateGrid/SelectableTemplateGrid.tsx [hunk 1]
```diff
@@ -51,19 +54,37 @@ function SelectableTemplateGrid<FormType extends FormWithTemplates>({
...
-    [updateTemplates, selectedTemplates],
+    [selectOrUnselectTemplate],
   );
 
+  useEffect(() => {
+    if (defaultSelectedTemplateKey) {
+      selectOrUnselectTemplate(defaultSelectedTemplateKey, 'select');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (austenem) [inline context/app/static/js/components/workspaces/SelectableTemplateGrid/SelectableTemplateGrid.tsx:86]: This is the `useEffect` issue mentioned above. I want this hook to be triggered on component mount only, which is why the empty dependency array is necessary - however, as this violates eslint rules, I'm assuming there is likely a better way to accomplish this. If the included dependencies are added to the array, the component will re-render infinitely.

**Review by john-conroy — COMMENTED** [review]
Can we add the default template key to `defaultValues.templates` in the workspace form hook?

**Comment by john-conroy** [comment]
If the above comment works, I don't think we would need to make any other change.

**Comment by NickAkhmetov** [comment]
I believe it should be possible to avoid using `useEffect`. Considering that the main goal is to set an initial selection, is there an approach that would allow us to define the initial state of the form to have this selected?

**Review by john-conroy — APPROVED** [review]
Nice, glad it worked!

**Comment by austenem** [comment]
Yes to both comments - that ended up being a much simpler solution!

**Pull request merged**